modules/ec2/iam.tf

```diff
@@ -0,0 +1,24 @@
+resource "aws_iam_role" "SSMRole" {
+  name = "SSMRole"
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": {
+      "Effect": "Allow",
+      "Principal": {"Service": "ssm.amazonaws.com"},
+      "Action": "sts:AssumeRole"
+   }
+ }
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "SSMRole-Attach" {
+  role       = aws_iam_role.SSMRole.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_instance_profile" "SSM-ASG" {
+  name = "SSM-ASG"
+  role = aws_iam_role.SSMRole.name
+}
```

Comment:
Please deploy this and try to launch a session in amazon session manager to make sure the configuration is correct